server/graphql/v2/query/collection/TransactionsCollectionQuery.ts

```diff
@@ -239,6 +245,31 @@ export const TransactionsCollectionResolver = async (
   }
 
   if (host) {
+    if (args.includeHost === false) {
+      const hostAccountsWithChildren = await fetchAccountsWithReferences(args.host, {
+        throwIfMissing: true,
+        attributes: ['id'],
+        include: [
+          {
+            association: 'children',
+            required: false,
+            attributes: ['id'],
+            where: { type: { [Op.ne]: CollectiveType.VENDOR } },
+          },
+        ],
+      });
+      const hostAccountsIds = uniq(
+        flatten(
+          hostAccountsWithChildren.map(account => {
+            const childrenIds = account.children?.map(child => child.id) || [];
+            return [account.id, ...childrenIds];
+          }),
+        ),
+      );
```

Comment:
I understand we're using this syntax elsewhere but reading this I wonder if this is optimal and could be replaced with a simpler account.getChildren

Reply:
It seems I am the original author 😱

Reply:
The main benefit is to trigger a single SQL query instead of two. Maybe not the simplest, but surely is the most optimal.

Reply:
As we're already fetching the host in another `fetchAccountWithReference` call, using `host.getChildren` here instead is also a single query, I've pushed a fix to use that instead.